File: java/push.java
import java.util.Stack;

class push extends Instruccion {
    private int valor;
    
    public push(int valor){
        this.valor = valor;
    }
    
    @Override
    public void ejecutar(Stack<Integer> pila){
        pila.push(valor); // Apilo el valor en la pila
    }
    
    @Override 
    public int incrementarPc(int pc){
        return pc + 1; // Incremento el contador de programa en 1
    }

    @Override
    public String mostrar(){
        return " push ";
    }
    
}

File: c++/push.cc
#include "push.h"
#include <iostream>

using namespace std;

push::push(int instruccionL) : instruccionL(instruccionL) { 
}

push::~push() {
}


void push::ejecutar(stack<int>& pila){
    pila.push(instruccionL);
}

int push::modificar(int pc){
    return pc + 1;
}

std::string push::mostrar(){
    return " push " + to_string(instruccionL);
}
